torch/csrc/jit/operator_upgraders/utils.cpp

```diff
@@ -0,0 +1,53 @@
+#include <torch/csrc/jit/operator_upgraders/utils.h>
+
+#include <c10/util/Optional.h>
+#include <torch/csrc/jit/operator_upgraders/version_map.h>
+#include <iostream>
+#include <regex>
+#include <string>
+#include <vector>
+
+namespace torch {
+namespace jit {
+
+c10::optional<UpgraderEntry> findUpgrader(
+    const std::vector<UpgraderEntry>& upgraders_for_schema,
+    size_t current_version) {
+  // we want to find the entry which satisfies following two conditions:
+  //    1. the version entry must be greater than current_version
+  //    2. Among the version entries, we need to see if the current version
+  //       is in the upgrader name range
+  auto pos = std::find_if(
+      upgraders_for_schema.begin(),
+      upgraders_for_schema.end(),
+      [current_version](const UpgraderEntry& entry) {
+        return entry.bumped_at_version > current_version;
+      });
+
+  if (pos != upgraders_for_schema.end()) {
+    return *pos;
+  }
+  return c10::nullopt;
+}
+
+bool isOpEntryCurrent(
```

Comment:
What is "OpEntry"? And what does it mean for an "OpEntry" to be "Current"? IIRC, based on our last discussion, only ops can be current.

Reply:
Yeah this is a bad naming. Will update

Reply:
done